python-backend/app/api/mines/incidents/resources/mine_incidents.py

```diff
@@ -119,70 +133,56 @@ def post(self, mine_guid):
-        type=str,
-        store_missing=False)
-    parser.add_argument(
-        'closing_report_summary',
-        help='Report from mine in reaction to the incident',
-        location='json',
-        type=str,
-        store_missing=False)
+    # required
+    parser.add_argument('incident_timestamp',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d %H:%M') if x else None,
+                        location='json',
+                        store_missing=False)
+    parser.add_argument('incident_description', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('reported_timestamp',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d %H:%M') if x else None,
+                        store_missing=False,
+                        location='json')
+    parser.add_argument('reported_by_name', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('reported_by_email', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('reported_by_phone_no', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('reported_by_phone_ext', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('emergency_services_called', type=inputs.boolean, location='json',
+                        store_missing=False)
+    parser.add_argument('number_of_injuries', type=int, location='json',
+                        store_missing=False)
+    parser.add_argument('number_of_fatalities', type=int, location='json',
+                        store_missing=False)
+    parser.add_argument('reported_to_inspector_party_guid', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('responsible_inspector_party_guid', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('determination_inspector_party_guid', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('determination_type_code', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('followup_investigation_type_code', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('followup_inspection', type=inputs.boolean, location='json',
+                        store_missing=False)
+    parser.add_argument('followup_inspection_date',
+                        type=lambda x: datetime.strptime(x, '%Y-%m-%d') if x else None,
+                        store_missing=False,
+                        location='json')
+    parser.add_argument('status_code', type=str, location='json',
+                        store_missing=False)
+    parser.add_argument('dangerous_occurrence_subparagraph_ids', type=list, location='json',
+                        store_missing=False)
+
```

Comment:
i wrote this an i don't like it. We should look more into using restplus models to define input.